### PR TITLE
ci: prepare npm scripts to support github actions pipeline

### PR DESCRIPTION
**Pull request by bardisg**
## Description of the change

This is part of series of PR to review changes for https://github.com/rudderlabs/rudder-sdk-js/pull/724

prepare npm scripts to support github actions pipeline

- moved all rollup configs in a folder
- renamed npm scripts to convention name:variant
- use --environment argument for rollup scripts
- moved babel config to browserlistand babelrc
- added a local dev server with livereload for local development
- use env files to manage write key in local dev sample app
- made eslint rules in line with CP team ones to align
- updated relevant documentation for build script usage
- added conventional commit check hook
- added comitizen (npm run commit) to assist with correct commit messages

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> [Fix [#1]() ](https://www.notion.so/rudderstacks/CI-CD-pipeline-JS-3d99ddf9b6f64501acb0cd567ced65ed)

### Development

- [ ] Lint rules pass locally
- [x] The code changed/added as part of this pull request has been covered with tests
- [x] All tests related to the changed code pass in development

### Code review 

- [x]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [x] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [x] Issue from task tracker has a link to this pull request

- - -
This change is [Reviewable](https://reviewable.io/reviews/rudderlabs/rudder-sdk-js/726)

### CODE REVIEW

**Comment by bardisg** [comment]
Size github check fails as production-staging does not have the script so action is not able to compare. 

No source code has changed and no extra size has been added in the bundles

File: .size-limit.js [hunk 1]
```diff
@@ -21,4 +21,10 @@ module.exports = [
     gzip: true,
     limit: '41 kB',
   },
+  {
+    name: 'Service Worker - NPM',
+    path: 'dist/rudder-sdk-js/service-worker/index.js',
+    gzip: true,
+    limit: '43 kB',
```
Comment (saikumarrs) [inline .size-limit.js:28]: How did it grow this huge?

File: package.json [hunk 2]
```diff
@@ -113,9 +139,23 @@
...
-    "*.js": "prettier --write"
+    "*.js": "eslint --cache --fix",
+    "*.{json,js,md,ts}": "prettier --write"
+  },
+  "config": {
+    "commitizen": {
+      "path": "./node_modules/@digitalroute/cz-conventional-changelog-for-jira",
+      "jiraMode": false,
+      "jiraOptional": true,
+      "skipScope": false,
```
Comment (saikumarrs) [inline package.json:157]: Let's keep the scope optional?

File: rollup-configs/integrationBuildScript.js [hunk 3]
```diff
@@ -1,6 +1,6 @@
 /* eslint-disable no-console */
```
Comment (saikumarrs) [inline rollup-configs/integrationBuildScript.js:1]: Let's move this file to a new directory (may be `scripts`) at the root level?

**Review by saikumarrs — APPROVED** [review]

**Comment by saikumarrs** [comment]
> Size github check fails as production-staging does not have the script so action is not able to compare.
> 
> No source code has changed and no extra size has been added in the bundles

The check fails even now. Let's change the base to the staging branch, bypass the checks and merge this.

File: package.json (hunk 2 above)
Comment (bardisg) [reply package.json:157]: Was thinking to have the scope as integration, core, service-worker but happy to start with optional for time being

File: rollup-configs/integrationBuildScript.js (hunk 3 above)
Comment (bardisg) [reply rollup-configs/integrationBuildScript.js:1]: sure will do this

File: .size-limit.js (hunk 1 above)
Comment (bardisg) [reply .size-limit.js:28]: now adheres to browserlist configs

**Review by saikumarrs — APPROVED** [review]

**Pull request merged**